src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -0,0 +1,215 @@
+ */
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr const auto& get(
+    const Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>& i) noexcept {
+  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
+
+  if constexpr (I == 0) return i.birth;
+  if constexpr (I == 1) return i.death;
+  if constexpr (I == 2) return i.dim;
+}
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::get` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
+ * @tparam dimension_type First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @tparam event_value_type Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @param i Interval from which the value should be returned.
+ * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
+ */
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr auto&& get(Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>&& i) noexcept {
+  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
+
+  if constexpr (I == 0) return std::move(i.birth);
+  if constexpr (I == 1) return std::move(i.death);
+  if constexpr (I == 2) return std::move(i.dim);
+}
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::get` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
+ * @tparam dimension_type First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @tparam event_value_type Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @param i Interval from which the value should be returned.
+ * @return Either the birth value if @p I == 0, the death value if @p I == 1 or the dimension if @p I == 2.
+ */
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr const auto&& get(
+    const Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>&& i) noexcept {
+  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
+
+  if constexpr (I == 0) return std::move(i.birth);
+  if constexpr (I == 1) return std::move(i.death);
+  if constexpr (I == 2) return std::move(i.dim);
+}
+
+}  // namespace persistence_matrix
+}  // namespace Gudhi
+
+namespace std {
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::tuple_size` for @ref Gudhi::persistence_matrix::Persistence_interval.
```

Comment:
"overload" is for functions, this is a (partial) specialization.

Reply:
Ok, I thought you could use it for both.